### PR TITLE
The new pipeline create flow should check for group admin as well

### PR DESCRIPTION
**Pull request by kritika-singh3**

### CODE REVIEW

File: spark/spark-spa/src/main/java/com/thoughtworks/go/spark/spa/PipelinesController.java [hunk 1]
```diff
@@ -45,7 +45,7 @@ public String controllerBasePath() {
     @Override
     public void setupRoutes() {
         path(controllerBasePath(), () -> {
-            before("", authenticationHelper::checkAdminUserAnd403);
+            before("", authenticationHelper::checkAdminUserOrGroupAdminUserAnd403);
```
Comment (maheshp) [inline spark/spark-spa/src/main/java/com/thoughtworks/go/spark/spa/PipelinesController.java]: No tests for this?

**Comment by kritika-singh3** [comment]
Related issue: #6475

File: spark/spark-spa/src/main/java/com/thoughtworks/go/spark/spa/PipelinesController.java (hunk 1 above)
Comment (marques-work) [reply spark/spark-spa/src/main/java/com/thoughtworks/go/spark/spa/PipelinesController.java]: Looks like the tests are there now in `spark/spark-spa/src/test/groovy/com/thoughtworks/go/spark/spa/PipelinesControllerTest.groovy`:

```groovy
    @Nested
    class Security implements SecurityTestTrait, GroupAdminUserSecurity { ... }
```

**Comment by marques-work** [comment]
@kritika-singh3 @maheshp Going to send this through the build and merge when green.

**Pull request merged**